### PR TITLE
Ingest enum payload from single-tenant aggregated endpoint

### PR DESCRIPTION
**Pull request by tilogaat**
*What*
Adds the entire pipeline for ingesting enums 

*How*
Adds a new rollup type EnumRollup, adds serialization-deserialization for Enums. With this code, we can ingest enums using the single tenant as follows:
```
curl -i -XPOST 'http://localhost:19000/v2.0/6076299/ingest/aggregated' -d '
{
    "tenantId": "5405532",
    "timestamp": 1438207529002,
    "enums": [
        {
            "name": "call_xyz_api",
            "value": "500"
        }
    ]
}'
```
You will observe that the enum call_xyz_api is stored in metrics_preaggregated_full 
The serialization format is the hashcode of the enum value followed by the count of the number of the times that value was seen. 

*Why*
This PR lays down the infrastructure to get enum metrics into the system. We can then add caching/validation logic of enums, rollups, queries, multi tenant ingestion on top of this.

### CODE REVIEW

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/Constants.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
     public static final byte VERSION_1_COUNTER_ROLLUP = 0;
     public static final byte VERSION_1_SET_ROLLUP = VERSION_1_ROLLUP; // don't change this.
```
Comment (goru97) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/Constants.java]: NIT: extra line.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/EnumRollup.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.rackspacecloud.blueflood.types;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EnumRollup implements Rollup {
+    private Map<Long, Long> en2Value = new HashMap<Long, Long>();
+
+    public EnumRollup withObject(Long valueNameHashcode, Long value) {
```
Comment (GeorgeJahad) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/EnumRollup.java]: Don't we need to store the full string name here, and not just the hash code?  Other wise how does the query endpoint return it?

Comment (GeorgeJahad) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/EnumRollup.java]: NIT: withObject() seems like a wierd name here.  What is the "object"?

File: blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/EnumRollupSerializationTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import java.io.*;
+import java.nio.ByteBuffer;
+
+public class EnumRollupSerializationTest {
+
+    @Test
+    public void testEnumV1RoundTrip() throws IOException {
+        EnumRollup e0 = new EnumRollup().withObject((long)"enumValue1".hashCode(), (long)1);
+        EnumRollup e1 = new EnumRollup().withObject((long)"t4.enum.abcdefg.hijklmnop.qrstuvw.xyz.ABCDEFG.HIJKLMNOP.QRSTUVW.XYZ.abcdefg.hijklmnop.qrstuvw.xyz.met".hashCode(), 34454722343L);
+
```
Comment (GeorgeJahad) [inline blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/EnumRollupSerializationTest.java]: When testing a container, (like EnumRollups), I prefer to see the test add at least two elements to the container.  Here you create 2 EnumRollups, with one element each.

**Comment by GeorgeJahad** [comment]
why are there no tests for enum rollups here:
https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/SerializationTest.java

https://github.com/rackerlabs/blueflood/blob/master/blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/IMetricSerializerTest.java

**Comment by tilogaat** [comment]
@GeorgeJahad SerializationTest.java has tests only for basic rollups. But I agree that those tests seem comprehensive. I will add a story to our backlog to add more comprehensive tests for aggregated metrics.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/NumericSerializer.java [hunk 4]
```diff
@@ -71,6 +66,7 @@
         static final byte B_TIMER = (byte)'T';
         static final byte B_SET = (byte)'S';
         static final byte B_GAUGE = (byte)'G';
+        static final byte B_ENUM = (byte) 'E';
```
Comment (VinnyQ) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/NumericSerializer.java]: NIT:  should remove the white space between the cast and the letter like the other constants above.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/NumericSerializer.java [hunk 5]
```diff
@@ -55,6 +48,7 @@
     public static AbstractSerializer<TimerRollup> timerRollupInstance = new TimerRollupSerializer();
     public static AbstractSerializer<SetRollup> setRollupInstance = new SetRollupSerializer();
     public static AbstractSerializer<GaugeRollup> gaugeRollupInstance = new GaugeRollupSerializer();
+    public static AbstractSerializer<EnumRollup> enRollupInstance = new EnumRollupSerializer();
```
Comment (VinnyQ) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/NumericSerializer.java]: NIT:  why not just spell out "enumRollupInstance"?  it's only 2 letters more and it reads clearer.

**Comment by VinnyQ** [comment]
:+1: looks good to me, @tilogaat !

**Pull request merged**